core/services/ocr2/delegate.go

```diff
@@ -111,8 +112,68 @@ func (d *Delegate) BeforeJobCreated(spec job.Job) {
 	// This is only called first time the job is created
 	d.isNewlyCreatedJob = true
 }
-func (d *Delegate) AfterJobCreated(spec job.Job)  {}
-func (d *Delegate) BeforeJobDeleted(spec job.Job) {}
+func (d *Delegate) AfterJobCreated(spec job.Job) {}
+func (d *Delegate) BeforeJobDeleted(jb job.Job) {
+	spec := jb.OCR2OracleSpec
+	if spec == nil {
+		d.lggr.Errorf("offchainreporting2.Delegate.BeforeJobDeleted called with wrong job type, ignoring non-OCR2 spec %v", jb)
+		return
+	}
+	if spec.Relay != relay.EVM {
+		return
+	}
+
+	chainID, err := spec.RelayConfig.EVMChainID()
+	if err != nil {
+		d.lggr.Errorf("OCR2 jobs spec missing chainID")
+		return
+	}
+	chain, err := d.chainSet.Get(big.NewInt(chainID))
+	if err != nil {
+		d.lggr.Error(err)
+		return
+	}
+	lp := chain.LogPoller()
+
+	var filters []string
+	switch spec.PluginType {
+	case job.OCR2VRF:
+		filters, err = ocr2vrfcoordinator.FilterNamesFromSpec(spec)
+		if err != nil {
+			d.lggr.Errorw("failed to derive ocr2vrf filter names from spec", "err", err, "spec", spec)
+		}
```

Comment:
Does mercury need a case here also?

Reply:
This PR calls UnregisterFilter() for all cases of RegisterFilter() in this repo, aside from the contract forwarder which is saved for a separate PR.

If Mercury has its own custom call to RegisterFilter, then we'll have to add a `FilterNamesFromSpec()` method to that in a Mercury PR before we can call it from this repo.  We'll need to do the same for CCIP and any other plugins that might have their own calls to RegisterFilter.

What I think we should have eventually is for `FilterNamesFromSpec()` to be a required method for all plugins to implement, and if they don't use logpoller then they just need a placeholder that returns empty list.  I was thinking we'd have it optional for now and just add it on a case by case basis, since other plugins like Functions don't even use logpoller and asking them to change the interface immediately along with this PR seems a bit much.

If it's easy to get a Mercury and/or CCIP PR in within the next few days, I could probably add those before merging this PR.  But it might be easier to handle those separately after these changes go in.  What do you think?

Reply:
Update:  you can disregard most of what I wrote above.  Looking through all the OCR2 plugin repos, I see that none of them besides ccip even imports logpoller, so there are no calls to RegisterFilter and therefore no need to make any changes to those repos.  CCIP is a special case that's outside the scope of this PR; it forks the entire logpoller so I think among other things, these changes will have to be backported over to that repo at some point.

Mercury doesn't need a case here, because its use of RegisterFilter is in the evm relay rather than the plugin as with vrf and keepers.  However, I do need to add it to the list of filters returned by `evmrelay.FilterNamesFromRelayArgs` here:
https://github.com/smartcontractkit/chainlink/pull/8787/files#diff-e6ef05d3b1ec007c275a6f1d690c96777f1b58ec80d080a53d63d694aed77e98R31-R138

Thanks for reminding me about Mercury!  Wasn't in there when I started working on this, so I missed it.

Reply:
Maybe at some point, we can move all of vrf & keepers' logpoller usage under the evm relay as well... if and when that happens, we can just get rid of this whole switch statement, and require that all future use of logpoller be in the relay.  That probably makes more sense with our architecture.

Reply:
```
services/relay/evm/mercury/config_poller.go
100:	err := destChainPoller.RegisterFilter(logpoller.Filter{Name: configFilterName, EventSigs: []common.Hash{FeedScopedConfigSet}, Addresses: []common.Address{addr}})
```

Do we need to care about this?